File: 2021/day01A.py
prev = None
count = 0

with open('day01_input.txt') as f:
    for line in f:
        curr = int(line.strip())
        if prev != None:
            count += 1 if curr > prev else 0

        prev = curr

print(f'Part A:\t{count}')

File: 2021/day01A.js
const fs = require('fs')
const inputFile = process.argv[2] || 'day01_input.txt'
let count = 0

const fileContents = fs.readFile(inputFile, 'utf8', (err, data) => {
    if (err) {
        throw err
    }

    const lines = data.split('\n')
    lines
        .map(n => Number.parseInt(n, 10))
        .reduce((prev, curr) => {
            if (curr > prev) {
                count += 1
            }

            return curr
        })
    console.log(count)
})
